pallets/parachain-staking/src/lib.rs

```diff
@@ -1338,6 +1370,80 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Hotfix to migrate a delegator's reserve to a lock. For any given delegator in the
+		/// provided list:
+		/// * this fn is idempotent
+		/// * is safe to call if the delegator doesn't exist
+		/// * is safe to call if the delegator has been migrated
+		/// * is safe to call if the delegator is a collator (this is a no-op)
+		///
+		/// weight calculation:
+		///   reads:
+		///    * DelegatorReserveToLockMigrations
+		///    * DelegatorState
+		///   writes:
+		///    * unreserve()
+		///    * set_lock()
+		///    * DelegatorReserveToLockMigrations
+		///   other: 50M flat weight + 100M weight per item
+		#[pallet::weight(
+			T::DbWeight::get().reads_writes(
+				2 * delegators.len() as u64,
+				3 * delegators.len() as u64
+			)
+			.saturating_add((delegators.len() as Weight).saturating_mul(100_000_000 as Weight))
+			.saturating_add(50_000_000 as Weight)
+		)]
+		pub fn hotfix_migrate_delegators_from_reserve_to_locks(
+			origin: OriginFor<T>,
+			delegators: Vec<T::AccountId>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			ensure!(delegators.len() < 100, <Error<T>>::InsufficientBalance);
```

Comment:
This error does not seem to represent the situation

Reply:
Also why 100? is this a safe bet based on benchmarks¿

Reply:
I think these values came from `hotfix_remove_delegation_requests_exited_candidates`. I don't want to introduce a new error variant just for this, but I'd take suggestions on using a different one.

`100` is definitely arbitrary, but it gives us some protection in case this can be abused somehow. Again, up for suggestions...

Reply:
Error changed in d2baeda